dimension/request.go

```diff
@@ -6,35 +6,50 @@ import (
 	"net/url"
 	"time"
 
+	"bytes"
+	"encoding/json"
 	"github.com/ONSdigital/go-ns/log"
 )
 
 // Request represents the request details
 type Request struct {
 	Attempt             int
-	Dimension           string
-	DimensionValue      string
+	DimensionID         string
+	Code                string
+	Value               string
+	CodeList            string
 	InstanceID          string
 	DatasetAPIURL       string
 	DatasetAPIAuthToken string
 	MaxAttempts         int
 }
 
+// DimensionOption to store in the dataset api
+type DimensionOption struct {
+	Name     string `json:"dimension_id"`
+	Code     string `json:"code"`
+	CodeList string `json:"code_list,omitempty"`
+	Value    string `json:"value"`
+}
+
 // Put executes a put request to the dataset API
-func (request *Request) Put(httpClient *http.Client) error {
+func (request *Request) Post(httpClient *http.Client) error {
 	// TODO Instead off backing off by bumping the sleep by 10 seconds per failed
 	// request, we may want an exponential backoff
 	time.Sleep(time.Duration(request.Attempt-1) * 10 * time.Second)
 
-	path := request.DatasetAPIURL + "/instances/" + request.InstanceID + "/dimensions/" + request.Dimension + "/options/" + request.DimensionValue
+	option, err := json.Marshal(DimensionOption{Name: request.DimensionID, Value: request.Value, CodeList: request.CodeList, Code: request.Code})
+
+	path := fmt.Sprintf("%s/instances/%s/dimensions", request.DatasetAPIURL, request.InstanceID)
+	fmt.Printf("path : %s\n", path)
```

Comment:
this should probably use the logger and provide more useful context about whats happening (or be removed)

Reply:
Sorry that some debug message I left in